src/components/navigation/tests/Navigation.spec.tsx

```diff
@@ -82,5 +83,14 @@ describe(' navigation', () => {
             navigation.setProps(newNavigationProps).update();
             expect(navigation.find(NavigationPerPage).props().currentPerPage).toBe(expectedPerPage);
         });
+
+        it('should call onPerPageClick prop with the correct values when it is set', () => {
+            const onPerPageClick = jasmine.createSpy('mockOnPerPageClick');
+            const perPageNumbers: number[] = [2, 3, 4];
+            const currentPerPage: number = 3;
+            const newNavigationProps: INavigationProps = _.extend({}, basicNavigationProps, {currentPerPage, perPageNumbers, onPerPageClick});
+            navigation.setProps(newNavigationProps).update().find(NavigationPerPageSelect).last().simulate('click');
+            expect(onPerPageClick).toHaveBeenCalledWith(4, 3, undefined); // newPerPage, currentPerPage, currentPage (currentPerPage and currentPage are undefined unless passed as props)

```

Comment:
Test looks god, I'd change the expect a bit:

```typescript
const currentPerPage = perPageNumbers[1];
const expectedPerPage = perPageNumbers[perPageNumber.length - 1];
...
expect(onPerPageClick).toHaveBeenCalledWith(expectedPerPage, currentPerPage, undefined);
```
This way if you add/modify the perPageNumbers const the test will stay green